src/ebonite/repository/metadata/local.py

```diff
@@ -310,9 +321,11 @@ def update_task(self, task: Task) -> Task:
         return task
 
     def delete_task(self, task: Task):
+        if task.models:
```

Comment:
I recommend using `self.get_models` here as `Task.models` is not guaranteed to reflect DB state.